### PR TITLE
wrote answer

### PR DESCRIPTION
**Pull request by Knguyen21**

### CODE REVIEW

File: public/diagnostic.js [hunk 1]
```diff
@@ -7,7 +7,33 @@
 //  -  a unit price
 //  -  a quantity
 //  -  a computed property called `orderPrice`, equal to price times quantity
-//
+
+const Order = Ember.Object.extend({
```
Comment (GA-MEB) [inline public/diagnostic.js:11]: This class definition is correct.

File: public/diagnostic.js [hunk 2]
```diff
@@ -7,7 +7,33 @@
...
+    unitPrice: '',
+    quantity: '',
+    orderPrice: Ember.computed('unitPrice', 'quantity', function(){
+      return this.get('unitPrice') * this.get('quantity');
+    })
```
Comment (GA-MEB) [inline public/diagnostic.js:16]: `orderPrice` is set up correctly.

File: public/diagnostic.js [hunk 3]
```diff
@@ -7,7 +7,33 @@
...
+});
+
```
Comment (MattB-GA-Boston) [inline public/diagnostic.js]: This class definition is also correct.

File: public/diagnostic.js [hunk 4]
```diff
@@ -7,7 +7,33 @@
...
+const Cart = Ember.Object.extend({
```
Comment (GA-MEB) [inline public/diagnostic.js:19]: This class definition is also correct.

File: public/diagnostic.js [hunk 5]
```diff
@@ -7,7 +7,33 @@
...
+  orders: [],
+  addToCart: function(){
+    return
+  }
+  totalPrice: Ember.computed('orderPrice')
```
Comment (GA-MEB) [inline public/diagnostic.js:24]: This is not the correct setup. A custom computed property needs a function to determine its value. Additionally, this property is watching `'orderPrice'`, which is not a property that a Cart has - it's a property of the Orders.

To monitor the array for changes, it should either watch `'orders'` or, if you want to monitor a property on an individual order, watch `orders.@each.propertyName`.

File: public/diagnostic.js [hunk 6]
```diff
@@ -7,7 +7,33 @@
...
+})
+
+let newCart = Cart.create({
+  orders: [
+    Ember.Object.create({quantity: 2, product: "hat!", price: 5}),
```
Comment (GA-MEB) [inline public/diagnostic.js:29]: These should be built using the Order class : `Order.create({})`

**Comment by GA-MEB** [comment]
4 / 7

---

**Breakdown**
-1 on `addToCart`
-1 on `totalPrice`
-1 for not instantiating the Order Ember Class

**Pull request closed without merging**